modules/build/src/main/scala/scala/build/Artifacts.scala

```diff
@@ -140,6 +147,30 @@ object Artifacts {
       )
     }
 
+    val fetchedScalaNativeCli = scalaNativeCliDependency match {
+      case Some(dependency) =>
+        Some(
+          value {
+            fetch(
+              Positioned.none(dependency),
+              Nil,
+              params,
+              logger,
+              Some(Set("_") ++ (if (fetchSources) Set("sources") else Set.empty)) // TODO not sure
```

Comment:
This isn't a dependency injected in the class path "seen" by users, no need to fetch sources.
```suggestion
              None
```